pkg/blockstorage/azure/azuredisk.go

```diff
@@ -117,6 +117,21 @@ func (s *adStorage) VolumeDelete(ctx context.Context, volume *blockstorage.Volum
 }
 
 func (s *adStorage) SnapshotCopy(ctx context.Context, from blockstorage.Snapshot, to blockstorage.Snapshot) (*blockstorage.Snapshot, error) {
+	return nil, errors.New("Copy Snapshot not implemented")
+}
+
+func (s *adStorage) SnapshotCopyWithArgs(ctx context.Context, from blockstorage.Snapshot, to blockstorage.Snapshot, args map[string]string) (*blockstorage.Snapshot, error) {
+	migrateStorageAccount := args[blockstorage.AzureMigrateStorageAccount]
+	if migrateStorageAccount == "" || args[blockstorage.AzureMigrateStorageKey] == "" {
```

Comment:
Why not also extract args[blockstorage.AzureMigrateStorageKey] to another variable? You de-reference it again below.